src/finch/symbolic/term.py

```diff
@@ -1,56 +1,65 @@
-from typing import Any, Iterator
-from abc import ABC, abstractmethod
-
 """
-This module contains definitions for common functions that are useful for symbolic expression manipulation.
-Its purpose is to provide a shared interface between various symbolic programming in Finch.
+This module contains definitions for common functions that are useful for symbolic
+expression manipulation. Its purpose is to provide a shared interface between various
+symbolic programming in Finch.
 
 Classes:
-    Term (ABC): An abstract base class representing a symbolic term. It provides methods to access the head
-    of the term, its children, and to construct a new term with a similar structure.
+    Term (ABC): An abstract base class representing a symbolic term. It provides methods
+    to access the head of the term, its children, and to construct a new term with a
+    similar structure.
 """
 
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Self
+
+__all__ = ["Term", "PreOrderDFS", "PostOrderDFS"]
+
+if TYPE_CHECKING:
+    from ..finch_logic import LogicNode
+
 
 class Term(ABC):
     def __init__(self):
         self._hashcache = None  # Private field to cache the hash value
 
     @abstractmethod
-    def head(self) -> Any:
```

Comment:
Let's keep it.

Reply:
Done in https://github.com/finch-tensor/finch-tensor-lite/pull/17/commits/56349b0d36760ac0f6f916d9b851a2d6744caddd.